src/main/java/com/aws/greengrass/shadowmanager/ShadowManager.java

```diff
@@ -320,23 +316,8 @@ private void configureSynchronization(Node newv) {
         }
     }
 
-    private void configureMaxSizeDocLimitConfig(Node newv) {
-        if (newv != null && !newv.childOf(CONFIGURATION_MAX_DOC_SIZE_LIMIT_B_TOPIC)) {
-            return;
-        }
-        int newMaxShadowSize = Coerce.toInt(config.lookup(CONFIGURATION_CONFIG_KEY,
-                CONFIGURATION_MAX_DOC_SIZE_LIMIT_B_TOPIC)
-                .dflt(DEFAULT_DOCUMENT_SIZE));
-        try {
-            Validator.validateMaxShadowSize(newMaxShadowSize);
-            Validator.setMaxShadowDocumentSize(newMaxShadowSize);
-            logger.atDebug()
-                    .setEventType("config")
-                    .kv("maxShadowSize", newMaxShadowSize)
-                    .log();
-        } catch (InvalidConfigurationException e) {
-            serviceErrored(e);
-        }
+    private void configureShadowDocSize(ShadowDocSizeConfiguration shadowDocSizeConfiguration) {
+        Validator.setMaxShadowDocumentSize(shadowDocSizeConfiguration.getMaxShadowDocSizeConfiguration());
```

Comment:
I know you didn't do this but this kind of validator object is less than ideal and not really object oriented. It has a mix of concerns and business logic that crosses boundaries... Ideally we had a Shadow class locally that did this... I think it is ok for now but we should look into fixing this.

The `Validator` is basically a "service object", those are generally bad because they break encapsulation. Here is nice article that talks about it [Why Service Objects Are a Bad Idea](https://www.codewithjason.com/rails-service-objects/?utm_source=pocket_reader)

> They’re chunks of procedural code masquerading as object-oriented code.

Reply:
Maybe add a TODO to fix this and stop using that class